addons/udes_delivery_control/models/stock_picking.py

```diff
@@ -0,0 +1,107 @@
+    u_delivery_note = fields.Char(related="u_extras_id.u_delivery_note")
+    u_location_id = fields.Many2one("stock.location", related="u_extras_id.u_location_id")
+    u_vehicle_scheduled_date = fields.Datetime(related="u_extras_id.u_vehicle_scheduled_date",)
+    u_vehicle_arrival_date = fields.Datetime(related="u_extras_id.u_vehicle_arrival_date",)
+    u_week = fields.Integer(related="u_extras_id.u_week")
+    u_unloading_start_date = fields.Datetime(related="u_extras_id.u_unloading_start_date")
+    u_unloading_end_date = fields.Datetime(related="u_extras_id.u_unloading_end_date")
+    u_time_taken = fields.Float(readonly=True, related="u_extras_id.u_time_taken")
+    u_vehicle_type = fields.Char(related="u_extras_id.u_vehicle_type")
+    u_trailer_number = fields.Char(related="u_extras_id.u_trailer_number")
+    u_lane_number = fields.Char(related="u_extras_id.u_lane_number")
+    u_pallet_count = fields.Integer(related="u_extras_id.u_pallet_count")
+    u_stillage_count = fields.Integer(related="u_extras_id.u_stillage_count")
+    u_box_count = fields.Integer(related="u_extras_id.u_box_count")
+    u_is_planned = fields.Boolean(related="u_extras_id.u_is_planned",)
+    u_is_late = fields.Boolean(related="u_extras_id.u_is_late",)
+    u_is_fast_track = fields.Boolean(related="u_extras_id.u_is_fast_track",)
+
+    # Backloading Fields
+    u_is_backload = fields.Boolean(related="u_extras_id.u_is_backload",)
+    u_backload_supplier = fields.Many2one(
+        "res.partner",
+        default=lambda self: self._default_u_supplier_id(),
+        related="u_extras_id.u_backload_supplier",
+    )
+    u_backload_pallet_count = fields.Integer(related="u_extras_id.u_backload_pallet_count",)
+    u_backload_stillage_count = fields.Integer(related="u_extras_id.u_backload_stillage_count",)
+    u_backload_cover_count = fields.Integer(related="u_extras_id.u_backload_cover_count",)
+    u_backload_reject_count = fields.Integer(related="u_extras_id.u_backload_reject_count",)
+    u_backload_start_date = fields.Datetime(related="u_extras_id.u_backload_start_date")
+    u_backload_end_date = fields.Datetime(related="u_extras_id.u_backload_end_date")
+    u_backload_time_taken = fields.Float(related="u_extras_id.u_backload_time_taken",)
+
+    def _create_picking_extras_data(self, values):
+        """ Create a transport information for each picking that doesn't
+            have it.
+        """
+        PickingExtras = self.env["udes_picking_extras"]
+        filtered_extras = dict(
+            (k[2:], values[k]) for k in EXTRA_FIELDS if k in values and values[k]
+        )
+        if filtered_extras:
+            for record in self:
+                if not record.u_extras_id:
+                    filtered_extras["picking_id"] = record.id
+                    PickingExtras.create(filtered_extras)
+
+        return values
+
+    @api.multi
+    def write(self, values):
+        values = self._create_picking_extras_data(values)
```

Comment:
This seems odd. 

We create and write to them as we would a normal field 
but then we access them via the extras, 

I'm not convinced that is a good idea. 
As we are trying to make them transparent but then we don't know where to find them.

Reply:
Wait I skipped over the related fields... is the much point of the extra model if we're just gonna relate them?

Reply:
Is this for searches?

Reply:
We're adding a new entity just for Delivery Control picking fields which is then linked to Delivery Control pickings only.

This means non-Delivery Control pickings don't get a load of irrelevant fields added.

We still want them to appear on the picking though, so that's why we have a load of related fields added.

Reply:
We need a widget for having subforms but I think that's been split off into an issue.